src/test/scala/com/advancedtelematic/campaigner/http/CampaignResourceSpec.scala

```diff
@@ -354,4 +356,96 @@ class CampaignResourceSpec
     }
   }
 
+  "GET /campaigns/:id/stats" should "return correct statistics" in {
+    val campaigns = Campaigns()
+
+    case class CampaignCase(
+        successfulDevices: Seq[DeviceId],
+        failedDevices: Seq[DeviceId],
+        cancelledDevices: Seq[DeviceId],
+        notAffectedCount: Long = Gen.choose[Long](0, 5).generate) {
+      val groupId = genGroupId.generate
+      val affectedDevices = successfulDevices ++ failedDevices ++ cancelledDevices
+      val affectedCount = affectedDevices.size.toLong
+      val processedCount = affectedCount + notAffectedCount
+      val groupStats = Stats(processedCount, affectedCount)
+
+      override def toString: String =
+        s"CampaignCase(s=${successfulDevices.size}, f=${failedDevices.size}, c=${cancelledDevices.size}, a=$affectedCount, n=$notAffectedCount, p=$processedCount)"
+    }
+
+    def conductCampaign(campaignId: CampaignId, campaign: CreateCampaign, campaignCase: CampaignCase): Future[Unit] = for {
+      _ <- campaigns.launch(campaignId)
+      _ <- campaigns.completeGroup(campaignId, campaignCase.groupId, campaignCase.groupStats)
+      _ <- campaigns.scheduleDevices(campaignId, campaign.update, campaignCase.affectedDevices:_*)
+      _ <- campaigns.finishDevices(campaignId, campaignCase.cancelledDevices, DeviceStatus.cancelled)
+      _ <- campaigns.finishDevices(campaignId, campaignCase.failedDevices, DeviceStatus.failed)
+      _ <- campaigns.finishDevices(campaignId, campaignCase.successfulDevices, DeviceStatus.successful)
+    } yield ()
+
+    Given("a main campaign that is finished")
+    val mainCase = CampaignCase(
+      successfulDevices = Gen.listOf(genDeviceId).generate,
+      failedDevices = Gen.nonEmptyListOf(genDeviceId).generate,
+      cancelledDevices = Gen.listOf(genDeviceId).generate
+    )
+    val (mainCampaignId, mainCampaign) = createCampaignWithUpdateOk(
+      genCreateCampaign().map(_.copy(groups = NonEmptyList.one(mainCase.groupId)))
+    )
+    conductCampaign(mainCampaignId, mainCampaign, mainCase).futureValue
+
+    When("requesting the stats for the main campaign")
+    Get(apiUri(s"campaigns/${mainCampaignId.show}/stats")).withHeaders(header) ~> routes ~> check {
+      status shouldBe OK
+
+      Then("the stats should be correct")
+      val campaignStats = responseAs[CampaignStats]
+      campaignStats.status shouldBe CampaignStatus.finished
+      campaignStats.finished shouldBe (mainCase.successfulDevices.size + mainCase.failedDevices.size)
+      campaignStats.failed should contain theSameElementsAs(mainCase.failedDevices)
```

Comment:
Just some silly style tip: the parenthesis after `theSameElementAs` are not necessary if you want.